docs/02-app/01-building-your-application/07-configuring/03-environment-variables.mdx

```diff
@@ -135,6 +135,32 @@ const env = process.env
 setupAnalyticsService(env.NEXT_PUBLIC_ANALYTICS_ID)
 ```
 
+### Runtime Environment Variables
+
+Next.js can support both build time and runtime environment variables.
+
+**By default, environment variables are only available on the server**. To expose an environment variable to the browser, it must be prefixed with `NEXT_PUBLIC_`. However, these public environment variables will be inlined into the JavaScript bundle during `next build`.
+
+To read runtime environment variables, we recommend using `getServerSideProps` or [incrementally adopting the App Router](/docs/app/building-your-application/upgrading/app-router-migration). With the App Router, we can safely read environment variables on the server during dynamic rendering. This allows you to use a singular Docker image that can be promoted through multiple environments with different values.
+
+```jsx
+import { unstable_noStore as noStore } from 'next/cache';
```

Comment:
I'm not sure that it's a good idea to advertise a clearly unstable API (prefixed with `unstable`) within a context of a "stable" feature (Runtime Environment Variables), when the `unstable_noStore` API itself doesn't have ANY docs whatsoever.

Reply:
> `unstable_noStore` is documented here : [nextjs.org/docs/app/api-reference/functions/unstable_noStore](https://nextjs.org/docs/app/api-reference/functions/unstable_noStore)

Yep, all good, saw this afterwards and edit my comment.

Reply:
It could also be another APIs, this is just the easiest way. e.g. could be `cookies()`, `headers()`, reading search params, any dynamic data fetch with `no-store`, for example.